### PR TITLE
docs: Updated README with more details

### PR DESCRIPTION
**Pull request by cluutch**
Please let me know if PRs like this are helpful @mvines. It took me some time to digest what is going on here. Will follow-up with some inline comments to the actual code, time permitting.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,56 @@
...
+## About
+`solana-cli-template` is a sample app demonstrating the creation of a minimal CLI application written in Rust to interact with Solana.
+It provides three pieces of functionality:
+
+- `ping`: Creates a transaction sending 0 SOL from the signer's account to the signer's account. Returns the signature of the transaction.
+- `balance`: Returns the account's balance.
+- `help`: Tips for using the app. This is an off-chain operation.
+
+## Local step-by-step
+1. Preparep account:
```
Comment (mvines) [inline README.md]:
```suggestion
1. Prepare account:
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,56 @@
...
+  - Start a local node: `solana test validator`.
+  - Generate a keypair: `solana-keygen new -o test.json`.
+  - Add 100 SOL to the corresponding account `solana airdrop --url http://127.0.0.1:8899 --keypair test.json 100`.
+
+2. Bulid app: `cargo run`.
```
Comment (mvines) [inline README.md]:
```suggestion
2. Build app: `cargo run`.
```

File: README.md [hunk 3]
```diff
@@ -1,2 +1,56 @@
...
+`solana-cli-template` is a sample app demonstrating the creation of a minimal CLI application written in Rust to interact with Solana.
+It provides three pieces of functionality:
+
+- `ping`: Creates a transaction sending 0 SOL from the signer's account to the signer's account. Returns the signature of the transaction.
+- `balance`: Returns the account's balance.
+- `help`: Tips for using the app. This is an off-chain operation.
+
+## Local step-by-step
+1. Preparep account:
+  - Start a local node: `solana test validator`.
```
Comment (mvines) [inline README.md]:
```suggestion
  - Start a local node: run `solana-test-validator`.
```

File: README.md [hunk 4]
```diff
@@ -1,2 +1,56 @@
...
+  - Generate a keypair: `solana-keygen new -o test.json`.
+  - Add 100 SOL to the corresponding account `solana airdrop --url http://127.0.0.1:8899 --keypair test.json 100`.
+
+2. Bulid app: `cargo run`.
+
+3. Ping:
+  ```
+  $ ./target/debug/cli-template ping --url http://127.0.0.1:8899 --keypair test.json
```
Comment (mvines) [inline README.md]:
```suggestion
  $ cargo run -- ping --url http://127.0.0.1:8899 --keypair test.json
```

File: README.md [hunk 5]
```diff
@@ -1,2 +1,56 @@
...
+  Signature: 2Y863JX96RTqbeGfcvQowVt1V91Dgs2LZfVgQ3mGJPmYu24sUTYmfkArHAAgj4uFqP75bm9GXU9DYjiMFxahQJUC
+  ```
+
+4. Balance:
+  ```
+  $ ./target/debug/cli-template balance --url http://127.0.0.1:8899 --keypair test.json
```
Comment (mvines) [inline README.md]:
```suggestion
  $ cargo run -- balance --url http://127.0.0.1:8899 --keypair test.json
```

File: README.md [hunk 6]
```diff
@@ -1,2 +1,56 @@
...
+  3dSRGE3wYCcGWFrxAsQs5PaBqtJzzxdTzY2ypXNFUji9 has a balance of ◎99.999995000 // balance less than 100 because of ping operation above
+  ```
+
+5. Run help for the complete list of options:
+  ```
+  $ ./target/debug/cli-template --help
```
Comment (mvines) [inline README.md]:
```suggestion
  $ cargo run -- -help
```

**Comment by mvines** [comment]
Thanks!  I made a couple suggestions and minor spelling fixes

File: README.md (hunk 1 above)
Comment (cluutch) [reply README.md]: Adopted: https://github.com/mvines/solana-cli-template/pull/2/commits/cbf1094b7ff393c9de85c85a124eea68dbb611b8#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R14

File: README.md (hunk 2 above)
Comment (cluutch) [reply README.md]: Adopted: https://github.com/mvines/solana-cli-template/pull/2/commits/cbf1094b7ff393c9de85c85a124eea68dbb611b8#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R19

File: README.md (hunk 3 above)
Comment (cluutch) [reply README.md]: Adopted: https://github.com/mvines/solana-cli-template/pull/2/commits/cbf1094b7ff393c9de85c85a124eea68dbb611b8#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R15

File: README.md (hunk 4 above)
Comment (cluutch) [reply README.md]: Cool, I didn't know this syntax. Adopted: https://github.com/mvines/solana-cli-template/pull/2/commits/cbf1094b7ff393c9de85c85a124eea68dbb611b8#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R23

File: README.md (hunk 5 above)
Comment (cluutch) [reply README.md]: Adopted: https://github.com/mvines/solana-cli-template/pull/2/commits/cbf1094b7ff393c9de85c85a124eea68dbb611b8#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R29

File: README.md (hunk 6 above)
Comment (cluutch) [reply README.md]: Adopted: https://github.com/mvines/solana-cli-template/pull/2/commits/cbf1094b7ff393c9de85c85a124eea68dbb611b8#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R35

**Comment by cluutch** [comment]
Thanks for the review!

**Comment by mvines** [comment]
lgtm

**Pull request merged**